### PR TITLE
add discord rich presence support

### PR DESCRIPTION
**Pull request by Jan200101**
support for display the current level will be added soon

### CODE REVIEW

**Comment by Titaniumtown** [comment]
Does this only support windows? I checked your changes. Shouldn't this also support linux as well?

**Comment by Jan200101** [comment]
It does support Linux out of the box
https://github.com/sm64pc/sm64pc/blob/f2d6df38d1a947e7c7e1b9bfe35cd72dfbb6a16b/Makefile#L405-L412

**Comment by Titaniumtown** [comment]
Oh sorry didn't see that, lol. I only saw the "Windows_build" bit, and I thought it was windows only, lol.

**Comment by Jan200101** [comment]
To be more clear:
It only supports Windows and Linux and 32 bit and 64 bit systems
I can add the Mac libs too but currently no one has ported sm64pc to macos

**Comment by Jan200101** [comment]
This needs testing on Windows

**Comment by harrisonlingren** [comment]
was trying to compile this on windows to test, but i'm getting errors while building. are there any special dependencies/flags/arguments needed aside from `DISCORDRPC=1`?

compile cmd: `~$ PATH=/mingw64/bin:/mingw32/bin:$PATH make DISCORDRPC=1`

build output:
```
C:/dev/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: build/us_pc/src/pc/discord/discordrpc.o: in function `InitializeDiscord':
C:\dev\msys64\home\sm64/src/pc/discord/discordrpc.c:142: undefined reference to `Discord_Initialize'
C:/dev/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: build/us_pc/src/pc/discord/discordrpc.o: in function `discordUpdateRichPresence':
C:\dev\msys64\home\sm64/src/pc/discord/discordrpc.c:252: undefined reference to `Discord_UpdatePresence'
C:/dev/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: build/us_pc/src/pc/discord/discordrpc.o: in function `discordShutdown':
C:\dev\msys64\home\sm64/src/pc/discord/discordrpc.c:257: undefined reference to `Discord_Shutdown'
C:/dev/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: build/us_pc/src/pc/discord/discordrpc.o: in function `discordReset':
C:\dev\msys64\home\sm64/src/pc/discord/discordrpc.c:282: undefined reference to `Discord_UpdatePresence'
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:781: build/us_pc/sm64.us.f3dex2e.exe] Error 1
```

**Comment by HengiFettlich** [comment]
@harrisonlingren you are missing the discord-rpc library, or it's not present in the expected directory.

**Comment by harrisonlingren** [comment]
@HengiFettlich no, it's in there: 

```
~$ ls src/pc/discord
> discordrpc.c  discordrpc.h  discord-rpc.lib  libdiscord-rpc.a
```

**Comment by Jan200101** [comment]
@harrisonlingren when you are compiling for Windows you also need to use WINDOWS_BUILD=1

**Comment by Jan200101** [comment]
the lastest commit should fix up any linking issues and allows me to build a valid binary using MXE but I suspect that MSYS2 might not be happy with it.
I am not able to confirm if the resulting windows binary actually enables discord rich presence because that is outside of my current scope of testing

**Comment by harrisonlingren** [comment]
> @harrisonlingren when you are compiling for Windows you also need to use WINDOWS_BUILD=1

Looks like building on a windows system automatically sets this flag?

`Makefile`:
```c
ifeq ($(OS),Windows_NT)
WINDOWS_BUILD := 1
endif
```

In any case, I ran `~$ make DISCORDRPC=1 WINDOWS_BUILD=1` and got the same result. It's possible it is a MSYS2 issue, but since that's the recommended tooling from the wiki that might be problematic...

**Comment by Jan200101** [comment]
I'm calling it quit here
If the Windows lib doesn't end up working I'm closing this and someone else can do it

File: Makefile [hunk 1]
```diff
@@ -620,7 +622,8 @@ $(BUILD_DIR)/src/game/ingame_menu.o: $(BUILD_DIR)/include/text_strings.h
 ifeq ($(DISCORDRPC),1)
 $(BUILD_DIR)/src/game/discord/discordrpc.o: $(BUILD_DIR)/include/text_strings.h
 endif
-endif
```
Comment (harrisonlingren) [inline Makefile:598]: Looks like removing this broke the build?

```
~$ make clean
Makefile:803: *** missing 'endif'.  Stop.
```

**Comment by Jan200101** [comment]
there

**Comment by coltongit** [comment]
Plz rebase this PR and change base to nightly @Jan200101

**Comment by Jan200101** [comment]
why

**Comment by coltongit** [comment]
Two reasons:
a) you have merge conflicts
b) nightly's where new merges are supposed to go to now. The other devs made that decision a few days ago

**Comment by Jan200101** [comment]
No one seemingly wants to do the required testing so I'll just leave the PR as is and not worry about any conflicts

**Comment by Jan200101** [comment]
I'll close this PR in the coming days since I do not have the things required to test on a Windows system with msys2 nor the time to setup a proper env

**Pull request closed without merging**